### PR TITLE
Click code in help views

### LINKED ISSUE #459: Help view: send code to console?
Since the new help view (v1.6.0) (which is great @ManuelHentschel) includes hyperlinks and highlighting, I was wondering if it would be technically possible to send code from the examples to the active R session?

### PR DESCRIPTION
**Pull request by ManuelHentschel**
Closes #459 #1137

Code sections in the help viewer are highlighted on hover, and can be clicked to be copied to the clipboard, or run in the terminal.

The behavior can be modified with the setting `"r.helpPanel.clickCodeExamples"`. Changes to this setting might require reloading the window to take effect.

### CODE REVIEW

File: src/helpViewer/panel.ts [hunk 1]
```diff
@@ -264,6 +265,18 @@ export class HelpPanel {
...
 			console.log(`Message (text): ${String(msg.text)}`);
+		} else if(msg.message === 'codeClicked') {
+			const code = String(msg.code || '');
+			console.log(`Code clicked: ${code}`);
+			const codeClickAction = config().get<CodeClickAction>('helpPanel.clickCodeExamples');
+			if(codeClickAction === 'Disabled' || !code){
+				// pass
+			} else if(codeClickAction === 'Copy'){
+				void vscode.env.clipboard.writeText(code);
+				void vscode.window.showInformationMessage(`Copied to clipboard: \`${code}\`.`);
```
Comment (renkun-ken) [inline src/helpViewer/panel.ts:276]: The message seems to look too messy in many cases:

image

File: package.json [hunk 2]
```diff
@@ -1464,6 +1464,21 @@
...
           "markdownDescription": "Show links to matching help pages in hover"
         },
+        "r.helpPanel.clickCodeExamples": {
+          "type": "string",
+          "default": "Copy",
+          "markdownDescription": "What happens when clicking code examples on help pages.",
+          "enum": [
+            "Disabled",
+            "Copy",
+            "Run"
```
Comment (renkun-ken) [inline package.json:1474]: It is nice to have the capability to either copying or running the example code. I wonder if it is possible to have both at the same time. For example, click to copy, and cmd+click to run, or something like this?

Comment (ManuelHentschel) [reply package.json]: Good idea. I wasn't sure how to handle the possible [modifier keys](https://github.com/reditorsupport/vscode-r/blob/23aa7ca821956e6dd86330e3b44789c800b28431/html/help/script.ts#L100-L103) best, especially w.r.t. mac (Should we just make them all configurable, or are there sensible "groupings"?).

**Review by renkun-ken — APPROVED** [review]
Works nicely! Thanks!

**Pull request merged**